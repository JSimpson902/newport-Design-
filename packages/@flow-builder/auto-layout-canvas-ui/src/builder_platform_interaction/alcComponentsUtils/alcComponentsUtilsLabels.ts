import faultConnectorBadgeLabel from '@salesforce/label/AlcConnector.faultConnectorBadgeLabel';
import afterLastBadgeLabel from '@salesforce/label/FlowBuilderConnectorLabels.afterLastBadgeLabel';
import forEachBadgeLabel from '@salesforce/label/FlowBuilderConnectorLabels.forEachBadgeLabel';
import branchHeadConnectorDescribedBy from '@salesforce/label/AlcConnector.branchHeadConnectorDescribedBy';
import straightConnectorDescribedBy from '@salesforce/label/AlcConnector.straightConnectorDescribedBy';
import branchHeadGoToConnectorDescribedBy from '@salesforce/label/AlcConnector.branchHeadGoToConnectorDescribedBy';
import goToConnectorDescribedBy from '@salesforce/label/AlcConnector.goToConnectorDescribedBy';
import postMergeGoToConnectorDescribedBy from '@salesforce/label/AlcConnector.postMergeGoToConnectorDescribedBy';
import postMergeConnectorDescribedBy from '@salesforce/label/AlcConnector.postMergeConnectorDescribedBy';
import loopCloseConnectorDescribedBy from '@salesforce/label/AlcConnector.loopCloseConnectorDescribedBy';
import branchHeadLoopCloseConnectorDescribedBy from '@salesforce/label/AlcConnector.branchHeadLoopCloseConnectorDescribedBy';
import postMergeLoopCloseConnectorDescribedBy from '@salesforce/label/AlcConnector.postMergeLoopCloseConnectorDescribedBy';
import emptyForEachDescribedBy from '@salesforce/label/AlcConnector.emptyForEachDescribedBy';
import ariaOnScheduledPathLabel from '@salesforce/label/AlcNode.ariaOnScheduledPathLabel';
import ariaOnPathImmediateLabel from '@salesforce/label/AlcNode.ariaOnPathImmediateLabel';
import ariaOnPathAfterLastLabel from '@salesforce/label/AlcNode.ariaOnPathAfterLastLabel';
import ariaOnFaultPathLabel from '@salesforce/label/AlcNode.ariaOnFaultPathLabel';
import ariaOnForEachPathLabel from '@salesforce/label/AlcNode.ariaOnForEachPathLabel';
import ariaEmptyBranchLabel from '@salesforce/label/AlcNode.ariaEmptyBranchLabel';
import airaOneGoToConnectorLabel from '@salesforce/label/AlcNode.airaOneGoToConnectorLabel';
import ariaMultiGoToConnectorLabel from '@salesforce/label/AlcNode.ariaMultiGoToConnectorLabel';
import ariaFollowedByPathImmediateLabel from '@salesforce/label/AlcNode.ariaFollowedByPathImmediateLabel';
import ariaRegularFollowedByLabel from '@salesforce/label/AlcNode.ariaRegularFollowedByLabel';
import ariaLoopFollowedByLabel from '@salesforce/label/AlcNode.ariaLoopFollowedByLabel';
import ariaGoToFollowedByLabel from '@salesforce/label/AlcNode.ariaGoToFollowedByLabel';
import ariaRegularPostMergeFollowedByLabel from '@salesforce/label/AlcNode.ariaRegularPostMergeFollowedByLabel';
import ariaLoopPostMergeFollowedByLabel from '@salesforce/label/AlcNode.ariaLoopPostMergeFollowedByLabel';
import ariaGoToPostMergeFollowedByLabel from '@salesforce/label/AlcNode.ariaGoToPostMergeFollowedByLabel';
import ariaOutcomeLabel from '@salesforce/label/AlcNode.ariaOutcomeLabel';
import ariaPauseConfigurationLabel from '@salesforce/label/AlcNode.ariaPauseConfigurationLabel';
import ariaOutcomeCountLabel from '@salesforce/label/AlcNode.ariaOutcomeCountLabel';
import ariaPauseConfigurationCountLabel from '@salesforce/label/AlcNode.ariaPauseConfigurationCountLabel';
import ariaScheduledPathCountLabel from '@salesforce/label/AlcNode.ariaScheduledPathCountLabel';
import ariaDecisionPathInfo from '@salesforce/label/AlcNode.ariaDecisionPathInfo';
import ariaDecisionPathGoToInfo from '@salesforce/label/AlcNode.ariaDecisionPathGoToInfo';
import ariaFaultPathLabel from '@salesforce/label/AlcNode.ariaFaultPathLabel';
import ariaFaultPathGoToLabel from '@salesforce/label/AlcNode.ariaFaultPathGoToLabel';
import ariaForEachPathLabel from '@salesforce/label/AlcNode.ariaForEachPathLabel';
import ariaForEachPathGoToLabel from '@salesforce/label/AlcNode.ariaForEachPathGoToLabel';
import ariaAfterLastPathLabel from '@salesforce/label/AlcNode.ariaAfterLastPathLabel';
import ariaAfterLastPathGoToLabel from '@salesforce/label/AlcNode.ariaAfterLastPathGoToLabel';
import ariaAfterLastPathLoopLabel from '@salesforce/label/AlcNode.ariaAfterLastPathLoopLabel';
import ariaPausePathInfo from '@salesforce/label/AlcNode.ariaPausePathInfo';
import ariaPausePathGoToInfo from '@salesforce/label/AlcNode.ariaPausePathGoToInfo';
import ariaScheduledPathInfo from '@salesforce/label/AlcNode.ariaScheduledPathInfo';
import ariaScheduledPathGoToInfo from '@salesforce/label/AlcNode.ariaScheduledPathGoToInfo';

export const LABELS = {
    ariaOnPathImmediateLabel,
    ariaOnPathAfterLastLabel,
    ariaEmptyBranchLabel,
    airaOneGoToConnectorLabel,
    ariaMultiGoToConnectorLabel,
    ariaFollowedByPathImmediateLabel,
    ariaRegularFollowedByLabel,
    ariaLoopFollowedByLabel,
    ariaGoToFollowedByLabel,
    ariaRegularPostMergeFollowedByLabel,
    ariaLoopPostMergeFollowedByLabel,
    ariaGoToPostMergeFollowedByLabel,
    ariaOutcomeLabel,
    ariaPauseConfigurationLabel,
    ariaOnScheduledPathLabel,
    ariaOutcomeCountLabel,
    ariaPauseConfigurationCountLabel,
    ariaScheduledPathCountLabel,
    ariaForEachPathLabel,
    ariaAfterLastPathLabel,
    ariaDecisionPathInfo,
    ariaPausePathInfo,
    ariaScheduledPathInfo,
    ariaFaultPathLabel,
    ariaFaultPathGoToLabel,
    ariaForEachPathGoToLabel,
    ariaAfterLastPathGoToLabel,
    ariaAfterLastPathLoopLabel,
    ariaScheduledPathGoToInfo,
    ariaDecisionPathGoToInfo,
    ariaPausePathGoToInfo,
    ariaOnFaultPathLabel,
    ariaOnForEachPathLabel,
    straightConnectorDescribedBy,
    goToConnectorDescribedBy,
    branchHeadConnectorDescribedBy,
    branchHeadGoToConnectorDescribedBy,
    postMergeConnectorDescribedBy,
    postMergeGoToConnectorDescribedBy,
    emptyForEachDescribedBy,
    loopCloseConnectorDescribedBy,
    branchHeadLoopCloseConnectorDescribedBy,
    postMergeLoopCloseConnectorDescribedBy,
    faultConnectorBadgeLabel,
    afterLastBadgeLabel,
    forEachBadgeLabel
};
