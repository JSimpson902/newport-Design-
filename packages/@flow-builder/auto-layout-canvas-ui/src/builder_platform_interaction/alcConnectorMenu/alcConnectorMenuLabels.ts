import actionSectionLabel from '@salesforce/label/AlcConnectorContextualMenu.actionSectionLabel';
import deleteGoToPathItemLabel from '@salesforce/label/AlcConnectorContextualMenu.deleteGoToPathItemLabel';
import goToPathItemLabel from '@salesforce/label/AlcConnectorContextualMenu.goToPathItemLabel';
import menuHeader from '@salesforce/label/AlcConnectorContextualMenu.menuHeader';
import pasteMultiItemLabel from '@salesforce/label/AlcConnectorContextualMenu.pasteMultiItemLabel';
import pasteOneItemLabel from '@salesforce/label/AlcConnectorContextualMenu.pasteOneItemLabel';
import reRouteGoToPathItemLabel from '@salesforce/label/AlcConnectorContextualMenu.reRouteGoToPathItemLabel';

export const labelsMap = {
    menuHeader,
    actionSectionLabel,
    pasteMultiItemLabel,
    pasteOneItemLabel,
    goToPathItemLabel,
    reRouteGoToPathItemLabel,
    deleteGoToPathItemLabel
};

export const LABELS: Labels<typeof labelsMap> = labelsMap;
