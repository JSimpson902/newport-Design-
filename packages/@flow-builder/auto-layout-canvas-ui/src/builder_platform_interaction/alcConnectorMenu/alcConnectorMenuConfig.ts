import { NodeType } from 'builder_platform_interaction/autoLayoutCanvas';
import { ICON_SHAPE } from 'builder_platform_interaction/alcComponentsUtils';
import { LABELS } from './alcConnectorMenuLabels';

import type { ElementMetadata, MenuSection } from 'builder_platform_interaction/autoLayoutCanvas';

import { storeUtils } from 'builder_platform_interaction/sharedUtils';

const { generateGuid } = storeUtils;

export const PASTE_ACTION = 'Paste';
export const GOTO_ACTION = 'goTo';
export const GOTO_REROUTE_ACTION = 'goToReroute';
export const GOTO_DELETE_ACTION = 'goToDelete';

const actionSection: MenuSection = {
    guid: generateGuid(),
    heading: '',
    items: [],
    label: LABELS.actionSectionLabel,
    separator: true
};

const pasteActionItem = {
    guid: generateGuid(),
    icon: 'utility:paste',
    iconContainerClass: 'slds-media__figure slds-listbox__option-icon',
    iconClass: '',
    iconSize: 'x-small',
    iconVariant: '',
    label: LABELS.pasteItemLabel,
    elementType: PASTE_ACTION,
    rowClass: 'slds-listbox__item action-row-line-height'
};

const addGoToActionItem = {
    guid: generateGuid(),
    icon: 'utility:level_down',
    iconContainerClass: 'slds-media__figure slds-listbox__option-icon',
    iconClass: '',
    iconSize: 'x-small',
    iconVariant: '',
    label: LABELS.goToPathItemLabel,
    elementType: GOTO_ACTION,
    rowClass: 'slds-listbox__item action-row-line-height'
};

const rerouteGoToActionItem = {
    guid: generateGuid(),
    icon: 'utility:level_down',
    iconContainerClass: 'slds-media__figure slds-listbox__option-icon',
    iconClass: '',
    iconSize: 'x-small',
    iconVariant: '',
    label: LABELS.reRouteGoToPathItemLabel,
    elementType: GOTO_REROUTE_ACTION,
    rowClass: 'slds-listbox__item action-row-line-height'
};

const deleteGoToActionItem = {
    guid: generateGuid(),
    icon: 'utility:delete',
    iconContainerClass: 'slds-media__figure slds-listbox__option-icon',
    iconClass: '',
    iconSize: 'x-small',
    iconVariant: '',
    label: LABELS.deleteGoToPathItemLabel,
    elementType: GOTO_DELETE_ACTION,
    rowClass: 'slds-listbox__item action-row-line-height'
};

/**
 * Create ALC menu configuration from the elements metadata
 *
 * @param elementsMetadata
 * @param showEndElement - Whether to show the end element item
 * @param isPasteAvailable - If paste is available
 * @param canAddGoto - Is the next element END
 * @param isGoToConnector - Is this a Goto connection
 */
export const configureMenu = (
    elementsMetadata: ElementMetadata[],
    showEndElement: boolean,
    isPasteAvailable: boolean,
    canAddGoto: boolean,
    isGoToConnector: boolean
) => {
    const sectionDefinitionsMap = {};

    let extraSections: MenuSection[] = [];
    actionSection.items = [];

    if (isPasteAvailable || canAddGoto || isGoToConnector) {
        if (isGoToConnector) {
            actionSection.items.push(rerouteGoToActionItem);
            actionSection.items.push(deleteGoToActionItem);
        } else if (canAddGoto) {
            actionSection.items.push(addGoToActionItem);
        }

        if (isPasteAvailable) {
            actionSection.items.push(pasteActionItem);
        }

        extraSections = [actionSection];
    }

    const sections = elementsMetadata.reduce(
        (
            acc,
            {
                section,
                description,
                icon,
                iconShape,
                iconBackgroundColor,
                label,
                elementType,
                elementSubtype,
                type,
                isSupported
            }
        ) => {
            if (section == null || (type === NodeType.END && !showEndElement)) {
                return acc;
            }

            let sectionDefinition = sectionDefinitionsMap[section];
            if (!sectionDefinition) {
                sectionDefinitionsMap[section] = sectionDefinition = {
                    guid: generateGuid(),
                    heading: type === NodeType.ORCHESTRATED_STAGE ? null : section,
                    label: section,
                    items: []
                };

                acc.push(sectionDefinition);
            }

            let iconContainerClass = 'slds-media__figure slds-listbox__option-icon';
            let iconClass = '';
            let iconSize = 'small';
            let iconVariant = '';

            if (iconBackgroundColor) {
                iconClass = iconBackgroundColor;
            }

            if (iconShape === ICON_SHAPE.CIRCLE && type === NodeType.END) {
                iconClass = `${iconClass} end-element-svg`;
                iconSize = 'xx-small';
                iconVariant = 'inverse';
            } else if (iconShape === ICON_SHAPE.DIAMOND) {
                iconContainerClass = `${iconContainerClass} rotate-icon-container slds-icon-standard-decision`;
                iconClass = `${iconClass} rotate-icon-svg`;
            }

            // Using the new isSupported property to determine what is shown in the connector menu
            if (isSupported) {
                const item = {
                    guid: generateGuid(),
                    description,
                    label,
                    elementType,
                    icon,
                    iconContainerClass,
                    iconClass,
                    iconSize,
                    iconVariant,
                    rowClass: 'slds-listbox__item',
                    elementSubtype
                };

                sectionDefinition.items.push(item);
            }

            return acc;
        },
        extraSections
    );
    // Filtering out sections that no longer have items in them.
    const updatedSections = sections.filter((section) => {
        return section.items?.length > 0;
    });
    return { sections: updatedSections };
};
