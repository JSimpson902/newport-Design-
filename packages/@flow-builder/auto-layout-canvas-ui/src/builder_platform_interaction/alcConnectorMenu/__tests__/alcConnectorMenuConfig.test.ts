// @ts-nocheck
import { ICON_SHAPE } from 'builder_platform_interaction/alcComponentsUtils';
import { NodeType } from 'builder_platform_interaction/autoLayoutCanvas';
import { ELEMENT_TYPE } from 'builder_platform_interaction/flowMetadata';
import { configureMenu, GOTO_DELETE_ACTION, GOTO_REROUTE_ACTION, PASTE_ACTION } from '../alcConnectorMenuConfig';
import { LABELS } from '../alcConnectorMenuLabels';

jest.mock('builder_platform_interaction/sharedUtils', () => {
    const mockSharedUtils = jest.requireActual('builder_platform_interaction_mocks/sharedUtils');
    return Object.assign({}, mockSharedUtils, {
        storeUtils: {
            generateGuid: jest.fn(() => 1)
        }
    });
});

const elementsMetadata = [
    {
        section: 'Interaction',
        type: 'default',
        icon: 'standard:screen',
        label: 'Screen',
        value: 'Screen',
        elementType: 'Screen',
        description: 'Collect information from'
    },
    {
        section: 'Interaction',
        type: 'default',
        icon: 'standard:custom_notification',
        label: 'Action',
        value: 'ActionCall',
        elementType: 'ActionCall',
        description:
            'Perform an action outside of the flow. Choose from your orgâ€™s custom create and update actions or an out-of-the-box action, like Post to Chatter or Submit for Approval.'
    },
    {
        section: 'Logic',
        type: 'branch',
        icon: 'standard:decision',
        iconShape: ICON_SHAPE.DIAMOND,
        label: 'Decision',
        value: 'Decision',
        elementType: 'Decision',
        description: 'Create Decision',

        actionType: 'testActionType',
        actionName: 'testActionName'
    },
    {
        section: 'Logic',
        type: 'end',
        icon: 'utility:stop',
        iconBackgroundColor: 'background-red',
        iconShape: ICON_SHAPE.CIRCLE,
        label: 'End',
        value: 'End',
        elementType: 'End',
        description: 'Create End'
    }
];

const metadata = {
    elementTypes: new Set(
        elementsMetadata.map(({ elementType }) => elementType).filter((elementType) => elementType !== 'ActionCall')
    ),
    isLoading: false,
    menuItems: [
        {
            guid: 'random_guid',
            description: 'Test Action with Static Resource',
            label: 'Action with Static Resource Icon',
            elementType: ELEMENT_TYPE.ACTION_CALL,
            actionType: 'apex',
            actionName: 'CustomSVGIconAction',
            actionIsStandard: false,
            icon: null,
            iconSrc: '/resource/1653591244000/customsvg#top',
            iconContainerClass: 'slds-media__figure slds-listbox__option-icon',
            iconClass: undefined,
            iconSize: 'small',
            iconVariant: '',
            rowClass: 'slds-listbox__item',
            elementSubtype: null,
            tooltip: 'Action with Static Resource Icon: Test Action with Static Resource',
            flowName: '',
            itemContainerClass: {
                'slds-media slds-listbox__option slds-listbox__option_entity': true,
                'slds-listbox__option_has-meta': true
            }
        },
        {
            guid: 'random_guid',
            description: '',
            label: 'Action with Slds Icon',
            elementType: ELEMENT_TYPE.ACTION_CALL,
            actionType: 'apex',
            actionName: 'ActionWithSldsIcon',
            actionIsStandard: false,
            icon: 'standard:feed',
            iconSrc: null,
            iconContainerClass: 'slds-media__figure slds-listbox__option-icon',
            iconClass: undefined,
            iconSize: 'small',
            iconVariant: '',
            rowClass: 'slds-listbox__item',
            elementSubtype: null,
            tooltip: 'Action with Slds Icon: Test Action with Slds Icon',
            flowName: '',
            itemContainerClass: {
                'slds-media slds-listbox__option slds-listbox__option_entity': true,
                'slds-listbox__option_has-meta': false
            }
        },
        {
            guid: 'random_guid',
            description: 'Post to the feed for a specific record, user, or Chatter group.',
            label: 'Post to Chatter',
            elementType: ELEMENT_TYPE.ACTION_CALL,
            actionType: 'chatterPost',
            actionName: 'chatterPost',
            actionIsStandard: true,
            icon: 'standard:custom_notification',
            iconSrc: undefined,
            iconContainerClass: 'slds-media__figure slds-listbox__option-icon',
            iconClass: 'background-navy',
            iconSize: 'small',
            iconVariant: '',
            rowClass: 'slds-listbox__item',
            elementSubtype: null,
            tooltip: 'Post to Chatter: Post to the feed for a specific record, user, or Chatter group.',
            flowName: '',
            itemContainerClass: {
                'slds-media slds-listbox__option slds-listbox__option_entity': true,
                'slds-listbox__option_has-meta': true
            }
        },
        {
            guid: 'random_guid',
            description: 'Flow to display the records',
            label: 'Display Records Flow',
            elementType: ELEMENT_TYPE.SUBFLOW,
            actionType: '',
            actionName: '',
            actionIsStandard: false,
            icon: 'standard:flow',
            iconSrc: undefined,
            iconContainerClass: 'slds-media__figure slds-listbox__option-icon',
            iconClass: 'background-navy',
            iconSize: 'small',
            iconVariant: '',
            rowClass: 'slds-listbox__item',
            elementSubtype: null,
            tooltip: 'Display the records in the flow.',
            flowName: 'DisplayRecords_Flow',
            itemContainerClass: {
                'slds-media slds-listbox__option slds-listbox__option_entity': true,
                'slds-listbox__option_has-meta': true
            }
        }
    ]
};

describe('connector menu config', () => {
    it('Match Snapshot with both paste and goto option', () => {
        expect(configureMenu('', metadata, elementsMetadata, false, 1, true, false)).toMatchSnapshot();
    });

    it('actionSection should have the right label', () => {
        expect(configureMenu('', metadata, elementsMetadata, false, 1, false, false).sections[0].label).toBe(
            LABELS.actionSectionLabel
        );
    });

    it('pasteItem should not be present when there are no available elements to paste', () => {
        expect(
            configureMenu('', metadata, elementsMetadata, false, 0, false, false).sections[0].items[0].elementType
        ).not.toBe(PASTE_ACTION);
    });

    it('pasteItem should have the right label when there is only 1 available element to paste', () => {
        expect(configureMenu('', metadata, elementsMetadata, false, 1, false, false).sections[0].items[0].label).toBe(
            LABELS.pasteOneItemLabel
        );
    });

    it('pasteItem should have the right label when there is more than 1 available element to paste', () => {
        expect(
            configureMenu('', metadata, elementsMetadata, false, 2, false, false).sections[0].items[0].label
        ).toEqual('AlcConnectorContextualMenu.pasteMultiItemLabel(2)');
    });

    it('pasteSection item should have the right icon', () => {
        expect(configureMenu('', metadata, elementsMetadata, false, 1, false, false).sections[0].items[0].icon).toBe(
            'utility:paste'
        );
    });

    it('pasteSection item should have the right rowClass', () => {
        expect(
            configureMenu('', metadata, elementsMetadata, false, 1, false, false).sections[0].items[0].rowClass
        ).toBe('slds-listbox__item action-row-line-height');
    });

    it('goToPath should have the right label', () => {
        expect(configureMenu('', metadata, elementsMetadata, false, 1, true, false).sections[0].items[0].label).toBe(
            `${LABELS.goToPathItemLabel}`
        );
    });

    it('goToPath item should have the right icon', () => {
        expect(configureMenu('', metadata, elementsMetadata, false, 1, true, false).sections[0].items[0].icon).toBe(
            'utility:level_down'
        );
    });

    it('goToPath item should have the right rowClass', () => {
        expect(configureMenu('', metadata, elementsMetadata, false, 1, true, false).sections[0].items[0].rowClass).toBe(
            'slds-listbox__item action-row-line-height'
        );
    });

    it('reRoute/delete should be displayed and GoTo should not with no pasteItem', () => {
        const menu = configureMenu('', metadata, elementsMetadata, false, 0, false, true);
        expect(menu.sections[0].items[0].label).toBe(LABELS.reRouteGoToPathItemLabel);
        expect(menu.sections[0].items[1].label).toBe(LABELS.deleteGoToPathItemLabel);
        expect(menu.sections[0].items[2]).toBe(undefined);
    });

    it('reRoute/delete should be displayed and GoTo should not with the right paste one item label', () => {
        const menu = configureMenu('', metadata, elementsMetadata, false, 1, false, true);
        expect(menu.sections[0].items[0].label).toBe(LABELS.reRouteGoToPathItemLabel);
        expect(menu.sections[0].items[1].label).toBe(LABELS.deleteGoToPathItemLabel);
        expect(menu.sections[0].items[2].label).toBe(LABELS.pasteOneItemLabel);
    });

    it('reRoute/delete should be displayed and GoTo should not with the right paste multi item label', () => {
        const menu = configureMenu('', metadata, elementsMetadata, false, 2, false, true);
        expect(menu.sections[0].items[0].label).toBe(LABELS.reRouteGoToPathItemLabel);
        expect(menu.sections[0].items[1].label).toBe(LABELS.deleteGoToPathItemLabel);
        expect(menu.sections[0].items[2].label).toBe('AlcConnectorContextualMenu.pasteMultiItemLabel(2)');
    });

    it('reRoutePath item should have the right icon', () => {
        expect(configureMenu('', metadata, elementsMetadata, false, 1, false, true).sections[0].items[0].icon).toBe(
            'utility:level_down'
        );
    });

    it('reRoutePath item should have the right rowClass', () => {
        expect(configureMenu('', metadata, elementsMetadata, false, 1, false, true).sections[0].items[0].rowClass).toBe(
            'slds-listbox__item action-row-line-height'
        );
    });

    it('deletePath item should have the right icon', () => {
        expect(configureMenu('', metadata, elementsMetadata, false, 1, false, true).sections[0].items[1].icon).toBe(
            'utility:delete'
        );
    });

    it('deletePath item should have the right rowClass', () => {
        expect(configureMenu('', metadata, elementsMetadata, false, 1, false, true).sections[0].items[1].rowClass).toBe(
            'slds-listbox__item action-row-line-height'
        );
    });

    it('search results should be correct based on search input', () => {
        const searchResultsMenu = configureMenu('action', metadata, elementsMetadata, false, 1, false, true);
        expect(searchResultsMenu.sections[0].items.length).toBe(2);
        // Highlight matched results.
        metadata.menuItems[0].formattedLabel = '<mark>Action</mark> with Static Resource Icon';
        metadata.menuItems[1].formattedLabel = '<mark>Action</mark> with Slds Icon';
        expect(searchResultsMenu.sections[0].items).toEqual(expect.arrayContaining([metadata.menuItems[0]]));
        expect(searchResultsMenu.sections[0].items).toEqual(expect.arrayContaining([metadata.menuItems[1]]));
        expect(searchResultsMenu.sections[0].items).toEqual(expect.not.arrayContaining([metadata.menuItems[2]]));
        delete metadata.menuItems[0].formattedLabel;
        delete metadata.menuItems[1].formattedLabel;
    });

    it('search results should be highlight partial matches', () => {
        const searchResultsMenu = configureMenu('tion', metadata, elementsMetadata, false, 1, false, true);
        expect(searchResultsMenu.sections[0].items.length).toBe(2);
        // Highlight matched results.
        metadata.menuItems[0].formattedLabel = 'Ac<mark>tion</mark> with Static Resource Icon';
        metadata.menuItems[1].formattedLabel = 'Ac<mark>tion</mark> with Slds Icon';
        expect(searchResultsMenu.sections[0].items).toEqual(expect.arrayContaining([metadata.menuItems[0]]));
        expect(searchResultsMenu.sections[0].items).toEqual(expect.arrayContaining([metadata.menuItems[1]]));
        expect(searchResultsMenu.sections[0].items).toEqual(expect.not.arrayContaining([metadata.menuItems[2]]));
        delete metadata.menuItems[0].formattedLabel;
        delete metadata.menuItems[1].formattedLabel;
    });

    it('search results should not show paste elements and go to connectors', () => {
        const searchResultsMenu = configureMenu('action', metadata, elementsMetadata, false, 1, false, true);
        expect(
            searchResultsMenu.sections[0].items.find((item) => item.elementType === GOTO_DELETE_ACTION)
        ).not.toBeTruthy();
        expect(
            searchResultsMenu.sections[0].items.find((item) => item.elementType === GOTO_REROUTE_ACTION)
        ).not.toBeTruthy();
        expect(searchResultsMenu.sections[0].items.find((item) => item.elementType === PASTE_ACTION)).not.toBeTruthy();
    });

    it('search results should list the subflows based on search input', () => {
        const searchResultsMenu = configureMenu('flow', metadata, elementsMetadata, false, 1, false, true);
        metadata.menuItems[3].formattedLabel = 'Display Records <mark>Flow</mark>';
        expect(searchResultsMenu.sections[0].items.length).toBe(1);
        expect(searchResultsMenu.sections[0].items).toEqual(expect.arrayContaining([metadata.menuItems[3]]));
        expect(searchResultsMenu.sections[0].items).toEqual(expect.not.arrayContaining([metadata.menuItems[0]]));
        expect(searchResultsMenu.sections[0].items).toEqual(expect.not.arrayContaining([metadata.menuItems[1]]));
        expect(searchResultsMenu.sections[0].items).toEqual(expect.not.arrayContaining([metadata.menuItems[2]]));
        delete metadata.menuItems[3].formattedLabel;
    });
});

describe('Section Heading', () => {
    it('is null when type is Orchestrator', () => {
        const metaDataWithOrchestrator = [
            {
                section: 'Logic',
                type: NodeType.ORCHESTRATED_STAGE,
                label: 'Stage',
                value: 'Orchestratedstage',
                elementType: 'Orchestratedstage'
            },
            {
                section: 'Logic',
                type: 'branch',
                label: 'Decision',
                value: 'Decision',
                elementType: 'Decision'
            }
        ];
        const menu = configureMenu('', metadata, metaDataWithOrchestrator, false, 1, false, true);
        expect(menu.sections.find((section) => section.label === 'Logic').heading).toBe(null);
    });

    it('is not null for other types', () => {
        const menu = configureMenu('', metadata, elementsMetadata, false, 1, false, true);
        expect(menu.sections.find((section) => section.label === 'Logic').heading).toBe('Logic');
    });

    it('is null when there is search input', () => {
        const menu = configureMenu('S', metadata, elementsMetadata, false, 1, false, true);
        expect(menu.sections[0].heading).toBe(null);
    });
});
