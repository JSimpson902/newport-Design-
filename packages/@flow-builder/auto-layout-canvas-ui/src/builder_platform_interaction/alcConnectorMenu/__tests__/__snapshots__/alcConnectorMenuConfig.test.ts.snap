// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`connector menu config Match Snapshot with both paste and goto option 1`] = `
Object {
  "sections": Array [
    Object {
      "guid": 1,
      "heading": "",
      "items": Array [
        Object {
          "elementType": "goTo",
          "guid": 1,
          "icon": "utility:level_down",
          "iconClass": "",
          "iconContainerClass": "slds-media__figure slds-listbox__option-icon",
          "iconSize": "x-small",
          "iconVariant": "",
          "label": "AlcConnectorContextualMenu.goToPathItemLabel",
          "rowClass": "slds-listbox__item action-row-line-height",
        },
        Object {
          "elementType": "Paste",
          "guid": 1,
          "icon": "utility:paste",
          "iconClass": "",
          "iconContainerClass": "slds-media__figure slds-listbox__option-icon",
          "iconSize": "x-small",
          "iconVariant": "",
          "label": "AlcConnectorContextualMenu.pasteItemLabel",
          "rowClass": "slds-listbox__item action-row-line-height",
        },
      ],
      "label": "AlcConnectorContextualMenu.actionSectionLabel",
      "separator": true,
    },
    Object {
      "guid": 1,
      "heading": "Interaction",
      "items": Array [
        Object {
          "description": "Collect information from",
          "elementSubtype": undefined,
          "elementType": "Screen",
          "guid": 1,
          "icon": "standard:screen",
          "iconClass": "",
          "iconContainerClass": "slds-media__figure slds-listbox__option-icon",
          "iconSize": "small",
          "iconVariant": "",
          "label": "Screen",
          "rowClass": "slds-listbox__item",
        },
      ],
      "label": "Interaction",
    },
    Object {
      "guid": 1,
      "heading": "Logic",
      "items": Array [
        Object {
          "description": "Create Decision",
          "elementSubtype": undefined,
          "elementType": "Decision",
          "guid": 1,
          "icon": "standard:decision",
          "iconClass": " rotate-icon-svg",
          "iconContainerClass": "slds-media__figure slds-listbox__option-icon rotate-icon-container slds-icon-standard-decision",
          "iconSize": "small",
          "iconVariant": "",
          "label": "Decision",
          "rowClass": "slds-listbox__item",
        },
      ],
      "label": "Logic",
    },
  ],
}
`;
